### PR TITLE
Bumping several github actions in Release workflow

### PR DESCRIPTION
**Pull request by mmartin24**
- Sync with https://github.com/epinio/ui-backend/pull/30
- Bump several outdated actions to prevent warnings as:
```
Node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: actions/checkout@v2, actions/setup-go@v2, docker/setup-buildx-action@v1, docker/login-action@v1. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.
```
Example: https://github.com/epinio/ui-backend/actions/runs/4755783459

## Description

~- Bumped var `SETUP_GO_VERSION`: from `1.18`  to  `1.20`~
- Bumped `actions/checkout` to `@v3`
- Bumped `actions/setup-go` to `@v4`. Added flag `cache: false` to prevent warnings as [this](https://github.com/epinio/ui-backend/actions/runs/5122380094) one when not finding `go.sum` file for cache
- Bumped `docker/setup-buildx-action` to `@v2`
- Bumped `login-action` to `@v2`

## Motivation and Context

Github actions will deprecate Node.js 12 actions soon and update to 16 was needed according to the warning:

```
node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: actions/checkout@v2, actions/setup-go@v2, docker/setup-buildx-action@v1, docker/login-action@v1. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.
```

## How Has This Been Tested?

Not tested because it would trigger a release

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Docs update
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Bump of dependencies

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have followed the guidelines in CONTRIBUTING.md, including the required formatting of the commit message

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
       - "v*"
 
 env:
-  SETUP_GO_VERSION: '1.18'
+  SETUP_GO_VERSION: '1.20'
```
Comment (richard-cox) [inline .github/workflows/release.yml]: This is just for something release related correct, and won't apply to the build of the ui-backend? That's currently still 1.18 - https://github.com/epinio/ui-backend/blob/main/src/jetstream/go.mod#LL3C1-L3C8

Comment (mmartin24) [reply .github/workflows/release.yml]: > This is just for something release related correct, and won't apply to the build of the ui-backend? That's currently still 1.18 - https://github.com/epinio/ui-backend/blob/main/src/jetstream/go.mod#LL3C1-L3C8

@richard-cox I am not sure. The idea was to be in sync with this other [pr](https://github.com/epinio/ui-backend/pull/30/files#diff-e0cfdbd18ccf2a8c9fb50fd963ecc670fb85a9cce968960b31d475b0d0263a1fR14) for the Release Next workflow. @enrichman do you foresee an impact on the build of the ui-backend? If unsure I can revert back to `1.18`

Comment (giubacc) [reply .github/workflows/release.yml]: maybe the ui-backend needs to be set to 1.20 to be consistent with all the other places (?)

Comment (mmartin24) [reply .github/workflows/release.yml]: thanks. I just returned it back to 1.18 for the moment. Perhaps we can re-take this conversation in a separate pr.

**Review by giubacc — APPROVED** [review]
lgtm

**Pull request merged**